libraries/SITL/SIM_StateSpace.cpp

```diff
@@ -0,0 +1,547 @@
+                        _servos_delayed_rp[i] = servos_raw[i];
+                    }
+            } else {
+                    pull_from_buffer_rp(_servos_delayed_rp);
+                    push_to_buffer_rp(servos_raw);
+            }
+
+            // this adds time delay for throttle inputs
+            if ((uint16_t)model.time_delay_col == 0 || is_zero(dt)) {
+                _servo_delayed_col = servos_raw[2];
+            } else if (servo_stored_col_buffer == nullptr) {
+                    uint16_t buffer_size = constrain_int16((uint16_t)model.time_delay_col, 1, 100) * 0.001f / dt;
+                    servo_stored_col_buffer = new ObjectBuffer<servo_stored>(buffer_size);
+                    while (servo_stored_col_buffer->space() != 0) {
+                        push_to_buffer_col(servos_raw);
+                    }
+                    _servo_delayed_col = servos_raw[2];
+            } else {
+                    pull_from_buffer_col(_servo_delayed_col);
+                    push_to_buffer_col(servos_raw);
+            }
+
+            // this adds time delay for pedal inputs
+            if ((uint16_t)model.time_delay_ped == 0 || is_zero(dt)) {
+                _servo_delayed_ped = servos_raw[3];
+            } else if (servo_stored_ped_buffer == nullptr) {
+                    uint16_t buffer_size = constrain_int16((uint16_t)model.time_delay_ped, 1, 100) * 0.001f / dt;
+                    servo_stored_ped_buffer = new ObjectBuffer<servo_stored>(buffer_size);
+                    while (servo_stored_ped_buffer->space() != 0) {
+                        push_to_buffer_ped(servos_raw);
+                    }
+                    _servo_delayed_ped = servos_raw[3];
+            } else {
+                    pull_from_buffer_ped(_servo_delayed_ped);
+                    push_to_buffer_ped(servos_raw);
+            }
+
+            // calculated motors class inputs from delayed servo_out for state space model input
+            float _roll_in = _servos_delayed_rp[0];  // roll
+            float _pitch_in = _servos_delayed_rp[1];  //pitch
+            float _throttle_in = _servo_delayed_col;  // throttle
+            float _yaw_in  = _servo_delayed_ped;  // yaw
+
+            if (is_zero(_throttle_in)) {
+            _pitch_in = 0;
+            _roll_in = 0;
+            _yaw_in = 0;
+            }
+
+            static float Dlonlag;
+            static float Dlatlag;
+            static float Dcollag;
+            static float Dpedlag;
+
+            // rotational acceleration (in rad/s/s?) in body frame
+            rot_accel.x = (model.Lv)*(velocity_air_bf.y)+(model.Llat)*(Dlatlag)+(model.Lp)*(gyro.x);
+            rot_accel.y = (model.Mu)*(velocity_air_bf.x)+(model.Mlon)*(Dlonlag);
+            rot_accel.z = (model.Nr)*(gyro.z)+((model.Nped)-(model.Lag*model.Lead))*(Dpedlag)+(model.Lag*model.Lead)*(_yaw_in);
+
+            float lateral_y_thrust = (model.Yv)*(velocity_air_bf.y)+(model.Ylat)*(Dlatlag);
```

Comment:
Perhaps run astyle on the code? Some operators have space around them, others don't.